core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationTabDirective.js

```diff
@@ -17,22 +17,273 @@
+          ngJoyrideTemplate = ngJoyrideTemplate.replace(
+            /\{\{/g, '<[').replace(/\}\}/g, ']>');
+          $templateCache.put('ng-joyride-title-tplv1.html',
+            ngJoyrideTemplate);
+
+          $scope.leaveTutorial = function() {
+            EditabilityService.onEndTutorial();
+            $scope.$apply();
+            StateTutorialFirstTimeService.markTranslationTutorialFinished();
+            $scope.translationTutorial = false;
+          };
+
+          $scope.onFinishTutorial = function() {
+            $scope.leaveTutorial();
+          };
+
+          $scope.onSkipTutorial = function() {
+            $scope.leaveTutorial();
+          };
+
+          $scope.translationTutorial = false;
+          $scope.onStartTutorial = function() {
+            if (GLOBALS.can_translate) {
+              $scope.translationTutorial = true;
+            }
+          };
+
+          $scope.showWelcomeTranslationModal = function() {
+            var modalInstance = $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/pages/exploration_editor/translation_tab/' +
+                'welcome_translation_modal_directive.html'),
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance', 'SiteAnalyticsService',
+                'ContextService',
+                function($scope, $uibModalInstance, SiteAnalyticsService,
+                    ContextService) {
+                  var explorationId = ContextService.getExplorationId();
+
+                  SiteAnalyticsService.registerTutorialModalOpenEvent(
+                    explorationId);
+
+                  $scope.beginTutorial = function() {
+                    SiteAnalyticsService.registerAcceptTutorialModalEvent(
+                      explorationId);
+                    $uibModalInstance.close();
+                  };
+
+                  $scope.cancel = function() {
+                    SiteAnalyticsService.registerDeclineTutorialModalEvent(
+                      explorationId);
+                    $uibModalInstance.dismiss('cancel');
+                  };
+
+                  $scope.editorWelcomeImgUrl = (
+                    UrlInterpolationService.getStaticImageUrl(
+                      '/general/editor_welcome.svg'));
+                }
+              ],
+              windowClass: 'oppia-welcome-modal'
+            });
+
+            modalInstance.result.then(function() {
+              $scope.onStartTutorial();
+            }, function() {
+              StateTutorialFirstTimeService.markTranslationTutorialFinished();
+            });
+          };
+
+          $rootScope.loadingMessage = 'Loading';
+          $scope.isTranslationTabBusy = false;
+          $scope.$on('refreshTranslationTab', function() {
+            $scope.$broadcast('refreshStateTranslation');
+          });
+          $scope.$on(
+            'enterTranslationForTheFirstTime',
```

Comment:
I think we can move this to the line above?

Reply:
Moved to the line above.